src/v/cluster/plugin_frontend.cc

```diff
@@ -0,0 +1,380 @@
+              }
+              if (!is_valid_utf8(v)) {
+                  return errc::transform_invalid_environment;
+              }
+          }
+
+          auto existing = _table->find_by_name(cmd.value.name);
+          if (existing.has_value()) {
+              // update!
+              // Only the offset pointer and environment can change.
+              if (existing->name != cmd.value.name) {
+                  return errc::transform_invalid_update;
+              }
+              if (existing->input_topic != cmd.value.input_topic) {
+                  return errc::transform_invalid_update;
+              }
+              if (existing->output_topics != cmd.value.output_topics) {
+                  return errc::transform_invalid_update;
+              }
+              return errc::success;
+          }
+
+          // create!
+          if (cmd.value.name().empty()) {
+              return errc::transform_invalid_create;
+          }
+          constexpr static size_t max_name_size = 128;
+          if (cmd.value.name().size() > max_name_size) {
+              return errc::transform_invalid_create;
+          }
+          if (!is_valid_utf8(cmd.value.name())) {
+              return errc::transform_invalid_create;
+          }
+          if (cmd.value.input_topic.ns != model::kafka_namespace) {
+              return errc::transform_invalid_create;
+          }
+          auto input_topic = _topics->get_topic_metadata(cmd.value.input_topic);
+          if (!input_topic) {
+              return errc::topic_not_exists;
+          }
+          const auto& input_config = input_topic->get_configuration();
+          if (input_config.is_internal()) {
+              return errc::transform_invalid_create;
+          }
+          // TODO: Support input read replicas?
+          if (input_config.is_read_replica()) {
+              return errc::transform_invalid_create;
+          }
+          if (cmd.value.output_topics.empty()) {
+              return errc::transform_invalid_create;
+          }
+          constexpr static size_t max_output_topics = 1;
+          if (cmd.value.output_topics.size() > max_output_topics) {
+              return errc::transform_invalid_create;
+          }
+          absl::flat_hash_set<model::topic_namespace> uniq(
+            cmd.value.output_topics.begin(), cmd.value.output_topics.end());
+          if (uniq.size() != cmd.value.output_topics.size()) {
+              return errc::transform_invalid_create;
+          }
+          for (const auto& out_name : cmd.value.output_topics) {
+              if (out_name.ns != model::kafka_namespace) {
+                  return errc::transform_invalid_create;
+              }
+              auto output_topic = _topics->get_topic_metadata(out_name);
+              if (!output_topic) {
+                  return errc::topic_not_exists;
+              }
+              const auto& output_config = output_topic->get_configuration();
+              if (output_config.is_internal()) {
```

Comment:
i guess at some point some logging or observability about these error cases would be useful?

Reply:
Ah very good idea. I've added logging for now.